cabal-install/tests/IntegrationTests2.hs

```diff
@@ -154,6 +160,13 @@ tests config =
       testCase "Test Config options for commented options" testConfigOptionComments,
       testCase "Test Ignore Project Flag" testIgnoreProjectFlag
     ]
+-- `haddock-project` is only supported by `haddock-2.26.1` and above which is
+-- shipped with `ghc-9.4`
+#if __GLASGOW_HASKELL__ >= 904
```

Comment:
CPP is a big red flag. I don't believe there's no other, more civil way to include a test only for newer compilers. If you can't find it in the old integration testsuite, I'd say, let's leave only the test in `cabal-testsuite` where we know how to check the compiler version.

Reply:
I thought I removed that test, I probably need to squash some commits.

Reply:
Actually the tests is still useful, I found a way to to avoid `CPP`.

Reply:
@coot I still see the CPP.

Reply:
Not any more :)